agent/agent_test.go

```diff
@@ -22,35 +22,35 @@ func TestAgent_OmitHostname(t *testing.T) {
 func TestAgent_LoadPlugin(t *testing.T) {
 	c := config.NewConfig()
 	c.InputFilters = []string{"mysql"}
-	err := c.LoadConfig("../internal/config/testdata/telegraf-agent.toml")
+	err := c.LoadConfig("../config/testdata/telegraf-agent.toml")
```

Comment:
Back on the topic of LoadConfig (since I can't reply to my previous comment anymore), what I'm really worried about is that this function has a lot of behavior.  I'm also sure we will want to modify it in the near future.  If we use this in the shim it may be difficult to guarantee the same handling of options.  It will also be quite permissive on what can be set, including things which won't work such as the interval, namepass/namedrop, global_tags, even the old single plugin per type Telegraf style, etc.

I think we would be better served with a simple toml parse to struct function.